auxiliary/proresdecoder.h

```diff
@@ -0,0 +1,148 @@
+    int             rowBytes;       //!< Number of bytes from first pixel of one line to first pixel of next line (must be at least PRBytesPerRowNeededInPixelBuffer() and a multiple of 16)
+    PRPixelFormat   format;         //!< Pixel format (2vuy, v210, v216, etc.)
+    int             width;          //!< Buffer width in pixels
+    int             height;         //!< Buffer height in pixels
+};
+typedef struct PRPixelBuffer PRPixelBuffer;
+
+
+/*!
+ * Utility routine that returns the minimum bytes per row needed in the
+ * destination pixel buffer in order to decode a frame of the specified width
+ * using the specified pixel format and downscale mode.
+ *
+ * @param frameWidth        Full-size frame width (not the downscaled width)
+ * @param pixelFormat       Pixel format of the pixel buffer
+ * @param downscaleMode     Downscaling mode
+ */
+int
+PRBytesPerRowNeededInPixelBuffer(
+        int frameWidth,
+        PRPixelFormat pixelFormat, 
+        PRDownscaleMode downscaleMode);
+
+
+/*!
+ * Opens a decoder instance and spawns "worker" threads.  If a
+ * threadStartupCallback callback function is supplied, each worker
+ * thread will call this function upon startup to provide a way for you
+ * to adjust the threads' priorities if necessary.
+ *
+ * @param numThreads    The number of simultaneous processing threads to spawn.
+ *                      Set this to 0 to have the decoder spawn according to
+ *                      the number of processors detected in the system.
+ * @param threadStartupCallback     An optional callback function for each
+ *                                  thread to call upon startup.  Set to NULL
+ *                                  if no callback is needed.
+ *
+ * @return A reference to the instantiated decoder, or NULL on failure.
+ */
+PRDecoderRef
+PROpenDecoder(int numThreads, void (*threadStartupCallback)());
+
+
+/*!
+ * Decodes a frame.  It is the caller's responsibility to ensure that
+ * destinationImage is a buffer of sufficient size and dimensions to contain
+ * the decoded image.
+ *
+ * @param decoder               A decoder instance returned by PROpenDecoder.
+ * @param compressedFrame       Pointer to the compresed frame.
+ * @param compressedFrameSize   Size of the compressed frame in bytes.
+ * @param destinationBuffer     Output buffer for the decoded image.
+ * @param downscaleMode         Requests decoder to perform downscaling.
+ * @param ignoreAlpha           If true, decoder will ignore encoded alpha
+ *                              channel data and output all opaque values.
+ *
+ * @return The number of bytes decoded for this frame, or a negative number in case of error.
+ */
+int
+PRDecodeFrame(
+        PRDecoderRef decoder,
+        const void* compressedFrame,
+        int compressedFrameSize,
+        PRPixelBuffer* destinationBuffer,
+        PRDownscaleMode downscaleMode,
+        bool ignoreAlpha);
```

Comment:
This needs the decoder reference, pointer to and the size of the compressed video frame, output buffer, downscale mode (always set to 1x1) and whether to consider the alpha (we always look for it) - that helps us define the context in our local header - `libavcodec/libproresdecoder.h::ProResDecodeCodec`